polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp

```diff
@@ -864,8 +864,12 @@ bool MemoryAccessMatrix::hasStridedAccessPattern(DataFlowSolver &solver) const {
         return false;
 
       if (isOnDiagonal) {
+        bool isFirstDiagonalElem = (row == 0 && col == 0);
```

Comment:
1 0 0
0 0 0
0 0 0
should also be considered as strided.
This is needed for `affine_3d` and `scf_3d` test cases in `loop_internalization.mlir`.